api/src/services/auth/user/user.resolver.ts

```diff
@@ -0,0 +1,178 @@
+    const p = generatePassword(context.app.get('settings').passwordPolicy)
+    return await passwordHash({ strategy: 'local' })(p, data, context)
+  },
+
+  resetExpires: async (resetExpires) => {
+    if (typeof resetExpires === 'number') {
+      return new Date(resetExpires).toISOString()
+    } else return resetExpires
+  },
+  verifyExpires: async (verifyExpires) => {
+    if (typeof verifyExpires === 'number') {
+      return new Date(verifyExpires).toISOString()
+    } else return verifyExpires
+  },
+
+  verifyChanges: async (verifyChanges) => {
+    if (typeof verifyChanges === 'string') {
+      return JSON.parse(verifyChanges)
+    } else return verifyChanges
+  },
+
+  createdAt: async (createdAt) => {
+    let stringToConvert: string | number | undefined = createdAt
+    if (!stringToConvert) stringToConvert = Date.now()
+    if (typeof stringToConvert === 'number') {
+      return new Date(stringToConvert).toISOString()
+    } else return createdAt
+  },
+
+  updatedAt: async (updatedAt) => {
+    let stringToConvert: string | number | undefined = updatedAt
+    if (!stringToConvert) stringToConvert = Date.now()
+    if (typeof stringToConvert === 'number') {
+      return new Date(stringToConvert).toISOString()
+    } else return updatedAt
+  },
+
+  lastConnection: async (lastConnection) => {
+    let stringToConvert: string | number | undefined = lastConnection
+    if (!stringToConvert) stringToConvert = Date.now()
+    if (typeof stringToConvert === 'number') {
+      return new Date(stringToConvert).toISOString()
+    } else return lastConnection
+  }
+})
+
+// Resolver for making partial updates
+export const userPatchResolver = resolve<UserPatch, HookContext>({
+  // schema: userPatchSchema,
+  validate: 'before',
+  properties: {
+    /**
+     * We clean the email
+     * from spaces and uppercase
+     * before inserting data in db.
+     */
+    email: async (email) => {
+      return email?.trim().toLowerCase()
+    },
+    // resetExpires: async (resetExpires) => {
+    //   if (typeof resetExpires === 'number') {
+    //     return new Date(resetExpires).toISOString()
+    //   }
+    // },
+    // verifyExpires: async (verifyExpires) => {
+    //   if (typeof verifyExpires === 'number') {
+    //     return new Date(verifyExpires).toISOString()
+    //   }
+    // },
+  },
+})
+
+// Resolver for the data that is being returned
+export const userResultResolver = resolve<UserResult, HookContext>({
+  // schema: userSchema,
+  validate: false,
+  properties: {
+    verifyChanges: async (verifyChanges) => {
+      if (typeof verifyChanges === 'string') {
+        return JSON.parse(verifyChanges)
+      }
+    },
+  },
+})
+
+// Resolver for the "safe" version that external clients are allowed to see
+export const userDispatchResolver: Resolver<UserResult, HookContext> = resolve<
+  UserResult,
+  HookContext
+>({
+  // schema: userSchema,
+  validate: false,
+  properties: {
+    isVerified: async () => undefined,
```

Comment:
Too many undefined ??? 
Is it related to the schema ?

Reply:
In fact, this is the dove way to remove properties before sending result to the end user (via REST API or WS, or another transport).
We set the property we don't want the user access to `undefined`, so they will not be sent.
Check this https://feathersjs.com/api/schema/resolvers.html#resolveexternal